azurerm/internal/services/compute/tests/linux_virtual_machine_resource_other_test.go

```diff
@@ -1573,3 +1613,75 @@ resource "azurerm_linux_virtual_machine" "test" {
 }
 `, template, data.RandomInteger, enabled)
 }
+
+func testLinuxVirtualMachine_otherSkipShutdownDefault(data acceptance.TestData) string {
+	template := testLinuxVirtualMachine_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_linux_virtual_machine" "test" {
+  name                       = "acctestVM-%d"
+  resource_group_name        = azurerm_resource_group.test.name
+  location                   = azurerm_resource_group.test.location
+  size                       = "Standard_F2"
+  admin_username             = "adminuser"
+  network_interface_ids = [
+    azurerm_network_interface.test.id,
+  ]
+
+  admin_ssh_key {
+    username   = "adminuser"
+    public_key = local.first_public_key
+  }
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+
+  source_image_reference {
+    publisher = "Canonical"
+    offer     = "UbuntuServer"
+    sku       = "16.04-LTS"
+    version   = "latest"
+  }
+}
+`, template, data.RandomInteger)
+}
+
+func testLinuxVirtualMachine_otherSkipShutdownDisabled(data acceptance.TestData) string {
+	template := testLinuxVirtualMachine_template(data)
+	return fmt.Sprintf(`
+%s
```

Comment:
instead we can specify this feature in the `provider` block, which in practice should be on every test rather than defined in the template - since the conditions can vary per-test

it's worth noting this won't work right now (insofar as all tests will run with the same provider) - but that's something that's being worked on - so if we can add this in preparation we're testing that going forward 👍

Reply:
Cool, I'm going to stop using the template and enable/disable the feature per-test.